### PR TITLE
Use an unintrusive approach to avoid to inherit from Orderable

### PR DESCRIPTION
**Pull request by DylannCordel**
Hi guys,

Thanks for the job already done ! Here is a PR to have a more flexible approach to be able, for exemple, to use this module with a thirdparty model we can not modify and already have an order field. It also manage signals to listen a reorder action. Sorry if there are many changes.

I also add a `get_sort_order_max` on the `Orderable` if, for exemple, we have `Book -> Chapter` and we want  to order `Chapter`. In that case, when we create a new chapter, the new order value won't be the max +1 of all existing chapters, but the max + 1 of same book's chapters. With the add of `get_sort_order_max`, we can still perform a super().save and just override get_sort_order_max with the expecting behaviour to get the next "max order value".

And there are also some fixes: 
- Fix keeping current filters when sorting was reset
- Fix class names (`get_extra_class_names_for_field_col` parameters were inverted)
- Fix CSS which had SCSS syntax
- fix the `fix_duplicate_positions` method. It reordered everithing with N+1 query and was not compatible with a "parent" model. Now, it just updates those who need to be and do it in 2 queries (first one to know if there are duplicates, second to reorder).
-  [minor non-backward compatible] the moved instance was updated with the save method, triggering signals of pre_save and post_save. But all other moved instances were updated via a SQL update (so no signals were sent). Now, everyone is updated via SQL and two signals are now triggered: `pre_reorder` and `post_reorder`, allowing people to know that some instances have been reordored (for exemple, to update an elastic search index used by vuestorefront). 

Tested with [wagtail-treemodeladmin](https://github.com/cfpb/wagtail-treemodeladmin) to be able to re-order `AdditionalContent` instances inside a django oscar `Product`.

### CODE REVIEW

**Comment by th3hamm0r** [comment]
+1

**Comment by elton2048** [comment]
Sorry for not maintaining the project for a long time. I hope the author or others are still active for this. There are some conflicts due to some simpler PRs I have reviewed and merged. It is good if some guys to resolve that and I will merge that.

File: wagtailorderable/modeladmin/mixins.py [hunk 1]
```diff
@@ -1,36 +1,97 @@
...
+    index_order method needs to be completed with an `admin_order_field` but as sort_order_field
+    is not yet known in the class, we need this meta class to get it from other final class args
+    """
+    def __new__(cls, name, bases, attrs):
+        model = attrs.get('model', None)
+        sort_order_field = attrs.get('sort_order_field', None)
+        if model and not sort_order_field:
+            sort_order_field = getattr(model, 'sort_order_field', None)
+        if sort_order_field:
+            # unfortunately, wagtail IndexView._get_default_ordering is curently using
```
Comment (elton2048) [inline wagtailorderable/modeladmin/mixins.py]: `currently`

File: wagtailorderable/modeladmin/mixins.py [hunk 2]
```diff
@@ -1,36 +1,97 @@
...
     """
+
     def __init__(self, parent=None):
         super(OrderableMixin, self).__init__(parent)
-        """
-        Don't allow initialisation unless self.model subclasses
-        `wagtail.wagtailcore.models.Orderable`
-        """
-        if not issubclass(self.model, Orderable):
+        # Don't allow initialisation unless self.model subclasses
```
Comment (elton2048) [inline wagtailorderable/modeladmin/mixins.py]: not developing a python project for a long time, can you explain why change the comment style here?

File: wagtailorderable/signal.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from django.dispatch import Signal
+
+# queryset is the whole "base queryset": it does not filter only on those updated
+# but filters on a parent key for exemple. eg. Book -> Chapter: Chapters need to be ordered
+# by book, and their ordering is important (can not be shared between all books)
+# in that case, you should use a "child model admin" which will allow you to display (and reorder)
+# only chapters for a specific books. The queryset sends to signals is filtered with the
+# "current book".
+
+pre_reorder = Signal(providing_args=['from_order', 'to_order', 'queryset'])
```
Comment (elton2048) [inline wagtailorderable/signal.py:10]: Seems `providing_args` is deprecated in Django 3.2. Could you update that to match the new version?

**Review by elton2048 — CHANGES_REQUESTED** [review]

**Comment by th3hamm0r** [comment]
@elton2048 thanks for your effort so far, I will try to update the PR (or create a new one) in the coming week.

File: wagtailorderable/signal.py (hunk 3 above)
Comment (th3hamm0r) [reply wagtailorderable/signal.py:10]: I've removed it and added comments instead, similar how wagtail does it in its code.

See #35

File: wagtailorderable/modeladmin/mixins.py (hunk 2 above)
Comment (th3hamm0r) [reply wagtailorderable/modeladmin/mixins.py]: I don't know either, maybe it was changed by mistake, I've changed it back it in #35

**Comment by th3hamm0r** [comment]
I've created a new, rebased PR #35

**Comment by th3hamm0r** [comment]
@elton2048 you can close this PR, since it is already merged and released with #35 (1.1.0)

**Pull request closed without merging**